reflect_test.go

```diff
@@ -331,3 +336,19 @@ func TestBaselineUnmarshal(t *testing.T) {
 
 	require.Equal(t, strings.ReplaceAll(string(expectedJSON), `\/`, "/"), string(actualJSON))
 }
+
+func TestSplitOnUnescapedCommas(t *testing.T) {
+	strToSplit := "Hello,this,is\\,a\\,string,haha"
```

Comment:
I'd define these examples with backticks (`), that way, they don't need to be double escaped and they more closely represent the use-case.